apiserver/facades/agent/uniter/networkinfo.go

```diff
@@ -13,26 +12,32 @@ import (
 	"github.com/juju/errors"
 	"github.com/juju/names/v4"
 	"github.com/juju/retry"
-	k8score "k8s.io/api/core/v1"
 
-	"github.com/juju/juju/apiserver/common"
 	"github.com/juju/juju/apiserver/params"
-	k8sprovider "github.com/juju/juju/caas/kubernetes/provider"
 	corenetwork "github.com/juju/juju/core/network"
 	"github.com/juju/juju/network"
 	"github.com/juju/juju/state"
 )
 
+type NetworkInfo interface {
+	ProcessAPIRequest(params.NetworkInfoParams) (params.NetworkInfoResults, error)
+	NetworksForRelation(
+		binding string, rel *state.Relation, pollPublic bool,
+	) (boundSpace string, ingress corenetwork.SpaceAddresses, egress []string, err error)
+
+	init(unit *state.Unit) error
```

Comment:
I'm not a fan of this.

```go
type NetworkInfoInit interface {
     init(unit *state.Unit) error
}
```

Then do a type check if an instance adheres to this.

```go
if v, ok := inst.(NetworkInfoInit); ok {
    return v.init(unit)
}
```

Reply:
I removed the `init` method. All the prep work is just done directly in `NewNetworkInfo`.